src/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -183,12 +199,51 @@ public bool IsSolutionOpen
                         await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                         return _dte != null &&
                                _dte.Solution != null &&
-                               _dte.Solution.IsOpen &&
-                               !IsSolutionSavedAsRequired();
+                               _dte.Solution.IsOpen;
                     });
             }
         }
 
+        public bool IsSolutionAvailable
+        {
+            get
+            {
+                return ThreadHelper.JoinableTaskFactory.Run(async delegate
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                    if (!IsSolutionOpen)
+                    {
+                        // Solution is not open. Return false.
+                        return false;
+                    }
+
+                    if (!IsSolutionSaveAsRequired())
+                    {
+                        // Solution is open and 'Save As' is not required. Return true.
+                        return true;
+                    }
+
+                    Init();
+                    var projects = _nuGetAndEnvDTEProjectCache.GetNuGetProjects().ToList();
+                    if (projects.Any(project => !(project is INuGetIntegratedProject)))
+                    {
+                        // Solution is open, but not saved. That is, 'Save as' is required.
+                        // And, there is a packages.config based project. Return false.
+                        return false;
+                    }
+
+                    // Solution is open and not saved. And, only contains project.json based projects.
+                    // Check if globalPackagesFolder is a full path. If so, solution is available.
+
+                    var settings = ServiceLocator.GetInstance<Configuration.ISettings>();
```

Comment:
As discussed with Justin offline, the singleton VS settings object depends on the singleton VS solution manager object. So, getting the instance every time is needed